File: Python Course/Software Development Module - in Python/Week11/Hairdresser.py
"""
Hairdresser.py
Billy Cussen
04/10/2020
"""

class myAppointment:
    def __init__(self,appointmentType, hairType):
        self.appointmentType = appointmentType
        self.hairType = hairType
        self.day = "Your Appointment is on: "

    def computeAppointmentDay(self):
        if(self.appointmentType =="style"):
            if(self.hairType == "long"):
                self.day += "Monday"
            elif(self.hairType == "mid"):
                self.day += "Tuesday"
            elif(self.hairType == "short"):
                self.day += "Wednesday"
            else:
                raise Exception("Something went Wrong here, invalid Hairtype selected!")
        elif(self.appointmentType == "cut"):
            if(self.hairType == "long"):
                self.day += "Thursday"
            elif(self.hairType == "mid"):
                self.day += "Friday"
            elif(self.hairType == "short"):
                self.day += "Saturday"
            else:
                raise Exception("Something went Wrong here, invalid Hairtype selected!")
        else:
            raise Exception("Something went wrong here, invalid Appointment Type selected!")

File: Java/Software Development Module - Java/Week11/Hairdresser.java
/*
Hairdresser.java
Billy Cussen
21/07/2020
*/

public class Hairdresser {
    
    //Data Members
    private String appointmentType;
    private String hairType;
    private String appointmentDay;

    //Constructor & Setter
    public Hairdresser(String appointmentType, String hairType){
        this.appointmentType = appointmentType;
        this.hairType = hairType;
        this.appointmentDay = "Your Appointment Day is: ";
    }

    //Compute & Getter
    public String computeAndGetAppointmentDay(){

        if(this.appointmentType.equalsIgnoreCase("Style")){
            if(this.hairType.equalsIgnoreCase("Long")){
                this.appointmentDay += "Monday";
            }else if(this.hairType.equalsIgnoreCase("Mid")){
                this.appointmentDay += "Tuesday";
            }else{
                this.appointmentDay += "Wednesday";
            }
        }else{
            if(this.hairType.equalsIgnoreCase("Long")){
                this.appointmentDay += "Thursday";
            }else if(this.hairType.equalsIgnoreCase("Mid")){
                this.appointmentDay += "Friday";
            }else{
                this.appointmentDay += "Saturday";
            }
        }

        return this.appointmentDay;
    }

}